### ISSUE TITLE
Should we add `quartodoc build` as a pre-render script in the suggested `_quarto.yml`? 

### ISSUE THREAD

**Issue by hamelsmu** [machow/quartodoc#141]
I am not sure if there is a way to bake this behavior in as an extension (I think there should be?), but in the absence of that I think the below config is a big quality of life improvement if using quartodoc

See https://quarto.org/docs/projects/scripts.html#pre-and-post-render

WDYT?

```yaml
project:
  type: website
  pre-render:
    - "quartodoc build"
```

**Comment by hamelsmu** [comment]
One thing is that an error in the pre-render script will not halt quarto from doing its thing, for example if `quartodoc build` exits with an error code 1, `quarto preview` will still do its thing.  Perhaps this is a bug?  Will discuss in slack

**Comment by machow** [comment]
Thanks for flagging--I definitely agree this is a big quality of life piece! (cf this issue about adding a `watch` command #114)

It seems like there might be surprising behaviors when using pre-render with `quarto preview`. For example, if we do run `quartodoc build` whenever [`QUARTO_PROJECT_RENDER_ALL` env variable](https://quarto.org/docs/projects/scripts.html#pre-and-post-render) is set, then we can rebuild everytime someone runs `quarto render`.

However, `quarto preview` won't trigger in this case:

* run `quarto preview` for the first time, let full site build
* it will trigger the pre-render with `QUARTO_PROJECT_RENDER_ALL`
* cancel out, then re-run `quarto preview`
* it **won't trigger** the pre-render with `QUARTO_PROJECT_RENDER_ALL`

I tested by adding `pre-render: 'bash -c "echo $QUARTO_PROJECT_RENDER_ALL"'`
(@cscheid do you know if this is intentional?)

**Comment by cscheid** [comment]
I don't know. @jjallaire designed the details of quarto preview vs pre-render scripts. (JJ, do you remember the reason for this behavior?)

**Comment by jjallaire** [comment]
With this change we always run the pre and post render scripts: https://github.com/quarto-dev/quarto-cli/commit/ade4a26148883a2b29bd0ca66142be645b460314

For the case of a preview where nothing needs to be rendered, you won't see `QUARTO_PROJECT_RENDER_ALL` but you will see `QUARTO_PROJECT_INPUT_FILES=""`.

**Comment by machow** [comment]
Ah thanks! We should be able to work with that! I can work on this once I pick up the `quartodoc watch` epic (#114), which likely won't be for another couple weeks. It seems like there may be a few trade-offs to consider, including increases in build time caused by triggering python (not necessarily a deal breaker, but I know it's a concern on the shiny team)

**Comment by machow** [comment]
Should be resolved by #210

**Comment by hamelsmu** [comment]
We can close this as this was resolved by #210
